solutions/USACO feb20-clocktree-silver.mdx

```diff
@@ -0,0 +1,109 @@
+---
+id: usaco-1016
+source: USACO Silver 2020 February
+title: Clock Tree
+author: Melody Yu
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_clocktree_silver_feb20.html)
+
+We can build a graph with the rooms as nodes and corriddors as edges. Consider a simple case: there are 2 nodes in the graph and one edge.
+Bessie can start from one of the rooms, go to its neighbor, increase the clock go back and increase the starting room's clock.
+She can go back and forth until one (or both) rooms' clock reaches 12. During this process, we noticed that the difference between two rooms' clock readings are not changed.
+If the starting clock readings differ by one, Bessie can start from the bigger clock reading room and travels back and forth until both of the clocks point to 12.
+If the starting clock readings differ by more than one, it is impossible for Bessie to make both clocks point to 12.
+
+If there are more nodes, we can treat it similar to 2 nodes by partition all the nodes in to 2 groups and make this problem a biparte graph problem.
+We can partition the nodes into two groups using DFS if 2 neighbor nodes belongs to different groups.
+
+Every time Bessie visits a neighbor node, the sum of clocks of the group that Bessie visits increase by 1.
+Then Bessie visits another neighbor node, the sum of clocks of the second group increase by 1.
+As both groups increase clock by one, the difference between the sum of clocks for these two groups does not change.
+This process repeats until clocks in 1 or 2 groups reache all 12.
+
+We can check the initial sum difference between two groups (mod 12) to figure out the final clock numbers.
+
+1. If group0 is same as group2 (mod 12), Bessie can start from any nodes
+2. If group0 is 1 smaller than group1 (mod 12), Bessie starts from any node of group1
+3. if group1 is 1 smaller than group0 (mod 12), Bessie starts from any node of group0
+4. If group0 and group1 differ by more than 2 (mod 12), there's no way to get all 12
```

Comment:
Doesn't the official solution already explain this well enough. Also, if you're going to put the video solution, you should remove this written solution.

Reply:
> Doesn't the official solution already explain this well enough?

based on this; no https://github.com/cpinitiative/usaco-guide/issues/870#issue-816921172

> Also, if you're going to put the video solution, you should remove this written solution.

some people prefer text, others prefer video so i think it's fine